buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java

```diff
@@ -211,4 +220,24 @@ private void assertFileProcessed(String testName) {
       + "}";
     assertTokenAssignmentIsParsed(declaration, assignment);
   }
+
+  @Test public void getFunctionNameSingleWord() {
+    String declaration = "String foo () :";
+    assertFunctionNameExtracted(declaration, "foo");
+  }
+
+  @Test public void getFunctionNameSingleAngleBrackets() {
```

Comment:
Should we have a test case with nested angle brackets? For example:

List<List<String>> foo ()

Reply:
Sure, will add.